flang/module/ieee_arithmetic.f90

```diff
@@ -462,8 +509,10 @@ end function ieee_real_a##AKIND##_i##KKIND;
   interface ieee_real
     SPECIFICS_I(IEEE_REAL_I)
     SPECIFICS_R(IEEE_REAL_R)
+#if FLANG_SUPPORT_R16
```

Comment:
These should still be enabled when libquadmath is linked: https://github.com/llvm/llvm-project/pull/102147/files#diff-b856c78ebb5a2b3c7fb791b499183206a28b6f7989d4c60fbbe9b8fbaaabe3ffR46

Reply:
I disabled these parts of the library when REAL16 is unsupported because we also disable REAL10 on unsupported machines. Is something different about REAL16 that means we should just build with warnings?